src/main/java/com/spikes2212/dashboard/AutoChooser.java

```diff
@@ -0,0 +1,189 @@
+     *               the name for the previously mentioned command
+     * @throws IllegalArgumentException when an even index is a {@link String}, an odd index is a {@link Command} or
+     * when the options' length isn't even.
+     */
+    public AutoChooser(Namespace namespace, Command defaultOption, String defaultOptionName, Object... options) {
+        this(namespace, defaultOption, defaultOptionName, INCREMENTATION_VALUE_TWO, options);
+    }
+
+    /**
+     * Creates an {@link AutoChooser} where the first {@link Command} and {@link String} will be used as the default
+     * option, and the rest of the options will be parsed as followed: every even index (0,2,4,...) will be a command
+     * and every odd index (1,3,5,...) will be used as the name for the previous command.
+     *
+     * @param defaultOption the default command this {@link AutoChooser} will use as the default option
+     * @param defaultOptionName the name for the default command
+     * @param options the rest of the options, where every even index is a command to be added and every odd index is
+     *               the name for the previous command
+     * @throws IllegalArgumentException when an even index is a {@link String}, an odd index is a {@link Command} or
+     * when the options' length isn't even.
+     */
+    public AutoChooser(Command defaultOption, String defaultOptionName, Object... options) {
+        this(new RootNamespace(DEFAULT_NAMESPACE_NAME), defaultOption, defaultOptionName, INCREMENTATION_VALUE_ONE, options);
+    }
+
+    private AutoChooser(Namespace namespace, Command defaultOption, String defaultOptionName,
+                        int incrementationValue, Object... options) {
+        if (options.length % 2 != 0) throw new IllegalArgumentException("The options' length isn't even.");
+        this.namespace = namespace;
+        setDefaultOption(defaultOptionName, defaultOption);
+        Command command;
+        String name;
+        int num;
+        for (int i = 0; i < options.length; i += 2) {
+            if (options[i] instanceof Command) {
+                command = (Command) options[i];
+            } else {
+                num = i + incrementationValue;
+                throw new IllegalArgumentException("The " + num + getCorrectSuffix(num) + " argument is not a Command.");
+            }
+            if (options[i + 1] instanceof String) {
+                name = (String) options[i + 1];
+            } else {
+                num = i + incrementationValue + 1;
+                throw new IllegalArgumentException("The " + num + getCorrectSuffix(num) + " argument is not a String.");
+            }
+            addOption(name, command);
+        }
+        putOnShuffleboard();
+    }
+
+    /**
+     * Adds the given command with the given name as an option to this {@link AutoChooser}. <br>
+     * In case this name already exists, a number will be added to differentiate between the options.
```

Comment:
When you use "in case" like this, it sounds like it does it either way **in case** it happens

Reply:
what?

Reply:
this sounds like a number is added either way in order to prepare for a case where name already exists